### ISSUE TITLE
Report error for explicit implementation of implicit IEquatable<T> interface on record type

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#45026]
An error should be reported for the explicit implementation of `IEquatable<R>`.
```C#
using System;

record R
{
    bool IEquatable<R>.Equals(R other) => false;
}
```

**Comment by AlekseyTs** [comment]
@cstone I think we decided that the code should be legal. Please re-activate if you disagree.
